packages/app/src/cli/models/app/app.ts

```diff
@@ -343,6 +359,45 @@ export function validateFunctionExtensionsWithUiHandle(
   return errors.length > 0 ? errors : undefined
 }
 
+export type UIExtensionType = zod.infer<typeof UIExtensionSchema>
+
+export function validateExtensionsHandlesInCollection(
+  editorExtensionCollections: ExtensionInstance<EditorExtensionCollectionType>[],
+  allExtensions: ExtensionInstance[],
+): string[] | undefined {
+  const errors: string[] = []
+
+  const allowableTypesForExtensionInCollection = ['ui_extension']
+  editorExtensionCollections.forEach((collection) => {
+    const extensionsInCollection = makeExtensionsInCollection(collection.configuration)
+
+    extensionsInCollection.forEach((extension) => {
+      const matchingExtension = findExtensionByHandle(allExtensions, extension.handle)
+
+      if (!matchingExtension) {
+        errors.push(
+          `[${collection.handle}] editor extension collection - Local app must contain an extension with handle '${extension.handle}'`,
+        )
+      } else if (!allowableTypesForExtensionInCollection.includes(matchingExtension.specification.identifier)) {
+        errors.push(
+          `[${collection.handle}] editor extension collection - The collection can't contain an extension of type '${matchingExtension.specification.identifier}'`,
+        )
+      } else if (matchingExtension.specification.identifier === 'ui_extension') {
+        const uiExtension = matchingExtension as ExtensionInstance<UIExtensionType>
+        uiExtension.configuration.extension_points.forEach((extensionPoint) => {
+          if (extensionPoint.target.startsWith('admin.')) {
```

Comment:
This feels a little hacky tbh, and should probably be done with the core validators.

Reply:
I dont think this is hacky because all the admin targets start with `admin.` that being said we decided to repeat these validations on the backend too

Reply:
Yeah as long we're doing that then it's all good. I thought this was going to be the only line of defence here.